core/src/test/scala/org/apache/spark/StatusTrackerSuite.scala

```diff
@@ -140,6 +140,14 @@ class StatusTrackerSuite extends SparkFunSuite with Matchers with LocalSparkCont
     }
 
     sc.removeJobTag("tag1")
+    eventually(timeout(60.seconds)) {
+      // Make sure that only two jobs are tagged after tag1
+      firstJobFuture.isCompleted should be(true)
+      secondJobFuture.isCompleted should be(true)
+      sc.getJobTags should be (Set("tag2"))
+      sc.statusTracker.getJobIdsForTag("tag1").toSet.size should be(2)
+    }
+
```

Comment:
This is not going to fix the issue - I have detailed the fix required for address the flakiness.